src/main/java/seedu/address/logic/parser/IsolatedEventCommandParser.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ENDDATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ISOEVENT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STARTDATE;
+
+import java.time.LocalDateTime;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.AddIsolatedEventCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.event.IsolatedEvent;
+
+/**
+ * Parser class for Isolated Event
+ */
+public class IsolatedEventCommandParser implements Parser<AddIsolatedEventCommand> {
+
+    /**
+     * Parse the IsolatedEvent command.
+     * @param args
+     * @return AddIsolatedEventCommand
+     * @throws ParseException
```

Comment:
Perhaps the Javadocs could include some description on what the parameters, return values and exceptions mean?